apps/total-typescript/src/hooks/use-mux-player.tsx

```diff
@@ -46,6 +44,22 @@ type VideoProviderProps = {
   path?: string
   muxPlayerRef: any
   onEnded?: () => Promise<any>
+  videoThumbId?: string
+}
+
+const useNextLesson = (
+  lesson: VideoResource,
+  module: SanityDocument,
+  section?: SanityDocument,
+) => {
+  const router = useRouter()
+  const {data: nextExercise} = trpc.lessons.getNextLesson.useQuery({
+    type: lesson._type,
+    slug: router.query.exercise as string,
+    module: module.slug.current,
+    section: section?.slug.current,
+  })
+  return nextExercise
 }
```

Comment:
this hook lets us load the next lesson async since we don't have the `solution` on the LessonResource